### PR TITLE
Bump phpunit version to 7.x

### PR DESCRIPTION
**Pull request by charescape**
#17698

### CODE REVIEW

**Comment by charescape** [comment]
php 7.4 failed...

image

File: tests/framework/db/CommandTest.php [hunk 1]
```diff
@@ -206,13 +206,13 @@ public function testBindParamValue()
         $row = $command->queryOne();
         $this->assertEquals($intCol, $row['int_col']);
         $this->assertEquals($charCol, $row['char_col']);
-        $this->assertEquals($floatCol, $row['float_col']);
+        $this->assertContains($row['float_col'], [$floatCol, sprintf('%.3f', $floatCol)]);
```
Comment (samdark) [inline tests/framework/db/CommandTest.php]:
```php
$this->assertEqualsWithDelta($floatCol, $row['float_col'], PHP_FLOAT_EPSILON);
```

**Comment by charescape** [comment]
@samdark  what about this one?

```php
// before:
$expectedMode = sprintf('%o', $expectedMode);

// after:
$expectedMode = str_pad(sprintf('%o', $expectedMode), '4', '0', STR_PAD_LEFT);
```
----------------------------------------
```
3) FileHelperTest::testCreateDirectory
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'700'
+'0700'
/home/travis/build/yiisoft/yii2/tests/framework/helpers/FileHelperTest.php:143
/home/travis/build/yiisoft/yii2/tests/framework/helpers/FileHelperTest.php:158
/home/travis/build/yiisoft/yii2/vendor/phpunit/phpunit/phpunit:61
```

**Comment by samdark** [comment]
Still checking the best way to fix it.

File: framework/base/Model.php [hunk 2]
```diff
@@ -14,6 +14,7 @@
 use ReflectionClass;
 use Yii;
 use yii\helpers\Inflector;
+use yii\helpers\StringHelper;
```
Comment (samdark) [inline framework/base/Model.php]: This import isn't needed.

Comment (charescape) [reply framework/base/Model.php]: done.

**Comment by samdark** [comment]
Preferred less invasive solution: https://github.com/yiisoft/yii2/pull/17725

**Comment by samdark** [comment]
Thanks for contributing this one. Gave me courage to continue digging and some important thoughts that led to final solution.

**Pull request closed without merging**